PSOperations/OperationCondition.swift

```diff
@@ -45,15 +42,27 @@ public protocol OperationCondition {
     func evaluateForOperation(_ operation: Operation, completion: @escaping (OperationConditionResult) -> Void)
 }
 
+public protocol AnyConditionError: Error {
```

Comment:
The name is IMHO necessary, because for the protocol `AnyConditionError`, there's no way to find out what condition it was. And we need a protocol that can be used as existential for e.g. `OperationConditionResult`. We also can't make the latter generic, since the type of the Error is not always known. Furthermore, then `OperationCondition` itself would need to become generic which would complicate a lot more things.

Reply:
D'oh! Forgot `OperationCondition` is a protocol